portal/common/lib/bcs_data_parsing.ts

```diff
@@ -2,20 +2,27 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { bcs, BcsType } from "@mysten/bcs";
-import { fromHEX, toHEX } from "@mysten/sui/utils";
+import { fromHEX, toHEX, toB64 } from "@mysten/sui/utils";
 import { base64UrlSafeEncode } from "./url_safe_base64";
 
 const Address = bcs.bytes(32).transform({
     input: (id: string) => fromHEX(id),
     output: (id) => toHEX(id),
 });
 
-// Blob IDs are represented on chain as u256, but serialized in URLs as URL-safe Base64.
+// Blob IDs & hashes are represented on chain as u256, but serialized in URLs as URL-safe Base64.
 const BLOB_ID = bcs.u256().transform({
     input: (id: string) => id,
     output: (id) => base64UrlSafeEncode(bcs.u256().serialize(id).toBytes()),
 });
 
+// Different than BLOB_ID, because we don't want this to be URL-safe;
+// otherwise, it will mess up with the checksum results.
+const U256 = bcs.u256().transform({
+    input: (id: string) => id,
+    output: (id) => toB64(bcs.u256().serialize(id).toBytes()),
+});
+
```

Comment:
Question: why B64? usually hashes are displayed in hex?

Reply:
Suggestion: keep a fixed-size byte array instead of a string. This would allow (1) not having the conversion to hex here, and (2) not having to covert the `sha256` hash to hex later on, simplifying the comparison.

Reply:
`toBytes` returns a `Uint8Array`. Uint8Array is a subclass of the hidden [TypedArray](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray) class.

There is no standard way of comparing two typed arrays, so we would have to compare them like so: 
```ts
function compareUint8Arrays(arr1, arr2) {
  if (arr1.length !== arr2.length) return false;
  return arr1.every((value, index) => value === arr2[index]);
}
```

or like this: 

```typescript
function compareUint8Arrays(arr1, arr2) {
  return Buffer.compare(Buffer.from(arr1), Buffer.from(arr2)) === 0;
}
```

On the other hand, with the existing implementation we can just compare 2 Base64 strings, which to me looks simpler.

Do we still need to use the Uint8Array approach? If so what would be the benefit?

Reply:
mmm I see. this is a very annoying language quirk?
Ok with strings then. But can we use Hex instead for the hash?

Reply:
Done ✔️